### PR TITLE
[JENKINS-32820, JENKINS-42164] - Windows service restart does not retain the build queue

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This is a merged version of #2019 from @atcarmo

- [x] - [JENKINS-32820](https://issues.jenkins-ci.org/browse/JENKINS-32820) Jenkins.cleanUp() was not called inside WindowsServiceLifecycle.restart()
- [x] - [JENKINS-42164](https://issues.jenkins-ci.org/browse/JENKINS-42164) Perform restart if the cleanup fails

> Basically Jenkins.cleanUp() is not being called inside the implementation of the restart method of WindowsServifeLifecycle. This caused, among other things, the queue not to be persisted on restart while Jenkins is installed as a Windows Service.

@jenkinsci/code-reviewers @daniel-beck  Please approve (especially the exception suppression logic)

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
OK, after merge I should always try to build the project...

**Review by daniel-beck — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Retriggering the test (the failure seems to be unrelated)

**Comment by oleg-nenashev** [comment]
Merging since restart should definitely proceed after the `contextDestroyed()` according to the code. Will create a placeholder issue

**Comment by oleg-nenashev** [comment]
Created https://issues.jenkins-ci.org/browse/JENKINS-42164 . Even taking the importance of the fix for Windows, I would rather vote against backporting, at least to the next .1 LTS. .2 may be fine

**Comment by oleg-nenashev** [comment]
Cannot squash from UI, because GitHub will keep me as a committer then

**Pull request merged**
